types/modules.d.ts

```diff
@@ -70,7 +70,130 @@ declare module 'time-ordered-set' {
 
   export = TimeOrderedSet
 }
-declare module 'hyperswarm'
+declare module 'hyperswarm' {
+  import EventEmitter from 'events'
+  import Dht from '@hyperswarm/dht'
+  import NoiseSecretStream from '@hyperswarm/secret-stream'
+
+  interface PeerDiscoveryOpts {
+    wait?: number | null
+    onpeer?: () => void
+    onerror?: (err: any) => void
+  }
+
+  export class PeerDiscovery {
+    readonly swarm: Hyperswarm
+    readonly topic: Buffer
+    isClient: boolean
+    isServer: boolean
+    destroyed: boolean
+    destroying: Promise<void> | null
+
+    constructor(swarm: Hyperswarm, topic: Buffer, opts?: PeerDiscoveryOpts)
+
+    session(opts: {
+      server?: boolean
+      client?: boolean
+      onerror?: (err: any) => void
+    }): PeerDiscoverySession
+    refresh(): Promise<void>
+    flushed(): Promise<boolean>
+    destroy(): Promise<void>
+  }
+
+  export class PeerDiscoverySession {
+    readonly discovery: PeerDiscovery
+    isClient: boolean
+    isServer: boolean
+    destroyed: boolean
+
+    constructor(discovery: PeerDiscovery)
+
+    get swarm(): PeerDiscovery['swarm']
+    get topic(): PeerDiscovery['topic']
+
+    refresh(opts?: { client?: boolean; server?: boolean }): Promise<void>
+    flushed(): Promise<boolean>
+    destroy(): Promise<void>
+  }
+
+  export class PeerInfo extends EventEmitter {
+    readonly publicKey: Buffer
+    readonly relayAddresses: any[]
+    reconnecting: boolean
+    proven: boolean
+    banned: boolean
+    tried: boolean
+    explicit: boolean
+    queued: boolean
+    client: boolean
+    topics: Buffer[]
+    attempts: number
+    priority: number
+
+    constructor(opts: { publicKey: Buffer; relayAddresses: any[] })
```

Comment:
slightly out of scope for this PR, but @sethvincent do you know what the type for the `relayAddresses` is?

Reply:
it looks like an array of objects with `host` and `port` properties based on hyperswarm/dht: https://github.com/hyperswarm/dht/blob/bb21a0191c1d96a727dcf82ffe3cb8967fb7c474/test/messages.js#L258

Reply:
ah nice! I'll update in a follow-up